frontend/packages/core/src/Network/errors.ts

```diff
@@ -0,0 +1,319 @@
+   */
+  description?: string;
+}
+
+/**
+ * Provides links to documentation or for performing an out of band action.
+ *
+ * For example, if a quota check failed with an error indicating the calling
+ * project hasn't enabled the accessed service, this can contain a URL pointing
+ * directly to the right place in the developer console to flip the bit.
+ */
+export interface Help {
+  /** URL(s) pointing to additional information on handling the current error. */
+  links?: {
+    /** Describes what the link offers. */
+    description?: string;
+    /**
+     * The URL of the link.
+     */
+    url?: string;
+  }[];
+}
+
+/**
+ * Provides a generic interface for error details. This should only be used if
+ * the error type cannot be mapped to a more specific type.
+ */
+export interface Details {
+  [key: string]: string;
+}
+
+/** An RPC error that occurred on the server. */
+export interface GrpcError extends Error {
+  /** The status code. */
+  code?: number;
+  /**
+   * A developer-facing error message in English. Any user-facing error messages
+   * will be sent in the details field.
+   */
+  message: string;
+  /** A list of objects that carry error details from the server. */
+  details?: (
+    | RetryInfo
+    | DebugInfo
+    | QuotaFailure
+    | ErrorInfo
+    | PreconditionFailure
+    | BadRequest
+    | RequestInfo
+    | ResourceInfo
+    | Help
+    | Details
+  )[];
+}
+
+/** An error received from the backend of Clutch. */
+export interface ClutchError extends GrpcError {
+  status: HttpStatus;
+}
+
+/**
+ * An error received from a HTTP request.
+ *
+ * This should be used for all network errors that occur
+ * talking to hosts other than the Clutch backend.
+ */
+export interface NetworkError extends Error {
+  status: HttpStatus;
+  message: string;
+  data: any;
+}
+
+/**
+ * An error received from the network client.
+ *
+ * For example, a request timeout due to client
+ * configuration having a timeout value too low.
+ */
+export interface ClientError {
+  message: string;
+}
+
+const GRPC_DETAIL_TYPES_MAP = {
+  "types.google.com/google.rpc.RetryInfo": {} as RetryInfo,
+  "types.google.com/google.rpc.DebugInfo": {} as DebugInfo,
+  "types.google.com/google.rpc.QuotaFailure": {} as QuotaFailure,
+  "types.google.com/google.rpc.ErrorInfo": {} as ErrorInfo,
+  "types.google.com/google.rpc.PreconditionFailure": {} as PreconditionFailure,
+  "types.google.com/google.rpc.BadRequest": {} as BadRequest,
+  "types.google.com/google.rpc.RequestInfo": {} as RequestInfo,
+  "types.google.com/google.rpc.ResourceInfo": {} as ResourceInfo,
+  "types.google.com/google.rpc.Help": {} as Help,
+};
+
+/**
+ * Construct a ClutchError from an AxiosError.
+ *
+ * @param clientError A client error object.
+ */
+const clutchError = (clientError: AxiosError) => {
+  const { data } = clientError?.response;
+
+  const error = {
+    code: data.code,
+    message: data.message,
+    status: grpcCodeToHttpStatus(data.code),
```

Comment:
can we rely on the http status code from the response instead? this could lead to a mismatch.